pkg/audit/audit.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"bytes"
+	"context"
+	"io"
+
+	"github.com/vivint/infectious"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/piecestore/rpc/client"
+)
+
+// Auditor struct
+type Auditor struct {
+	ps client.PSClient
+}
+
+// NewAuditor creates a new instance of an Auditor struct
+func NewAuditor(ps client.PSClient) *Auditor {
+	return &Auditor{ps: ps}
+}
+
+func (a *Auditor) getShare(ctx context.Context, stripeIndex, shareSize, stripeNumber int,
+	id client.PieceID, pieceSize int64) (share infectious.Share, err error) {
+
+	rr, err := a.ps.Get(ctx, id, pieceSize, &pb.PayerBandwidthAllocation{})
+	if err != nil {
+		return infectious.Share{}, err
+	}
+
+	offset := shareSize * stripeIndex
+
+	rc, err := rr.Range(ctx, int64(offset), int64(shareSize))
+	if err != nil {
+		return infectious.Share{}, err
+	}
+
+	buf := make([]byte, shareSize)
+	_, err = io.ReadFull(rc, buf)
+	if err != io.EOF || err != io.ErrUnexpectedEOF {
```

Comment:
This expression always evaluates to `true` no matter the value of `err`. This check should be as simple as `if err != nil {`.